utilities/simulator_cache/cache_simulator.h

```diff
@@ -38,22 +61,102 @@ class CacheSimulator {
   CacheSimulator& operator=(CacheSimulator&&) = delete;
 
   virtual void Access(const BlockCacheTraceRecord& access);
-  void reset_counter() { sim_cache_->reset_counter(); }
-  double miss_ratio();
-  uint64_t total_accesses();
+  void reset_counter() {
+    num_misses_ = 0;
+    num_accesses_ = 0;
+    user_accesses_ = 0;
+    user_misses_ = 0;
+  }
+  double miss_ratio() {
+    if (num_accesses_ == 0) {
+      return -1;
+    }
+    return static_cast<double>(num_misses_ * 100.0 / num_accesses_);
+  }
+  uint64_t total_accesses() { return num_accesses_; }
+
+  double user_miss_ratio() {
+    if (user_accesses_ == 0) {
+      return -1;
+    }
+    return static_cast<double>(user_misses_ * 100.0 / user_accesses_);
+  }
+  uint64_t user_accesses() { return user_accesses_; }
 
  protected:
-  std::shared_ptr<SimCache> sim_cache_;
+  void UpdateMetrics(bool is_user_access, bool is_cache_miss);
+
+  std::unique_ptr<GhostCache> ghost_cache_;
+  std::shared_ptr<Cache> sim_cache_;
```

Comment:
In the long term, do you plan to change `sim_cache_` from `SimCache` to `Cache`?

Reply:
We should use Cache.